### PR TITLE
config/jobs: use k8s-staging-test-infra images for some jobs

### PR DESCRIPTION
**Pull request by spiffxp**
Related:
- Part of: https://github.com/kubernetes/k8s.io/issues/1523
- Followup to: https://github.com/kubernetes/test-infra/pull/23656

### CODE REVIEW

File: config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml [hunk 1]
```diff
@@ -904,7 +904,7 @@ periodics:
     path_alias: k8s.io/kubernetes
   spec:
     containers:
-    - image: gcr.io/k8s-testimages/benchmarkjunit:latest
+    - image: gcr.io/k8s-staging-test-infra/benchmarkjunit:latest
```
Comment (ameukam) [inline config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml:907]: There is not tag for this image. Do we still want to do the flip ?

Comment (spiffxp) [reply config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml:907]: Good catch, I'm looking into why the image is missing...

Comment (spiffxp) [reply config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml:907]: Opened https://github.com/kubernetes/test-infra/pull/23771 to address

I'm going to use `gcrane` to manually sync something over in the meantime

Comment (spiffxp) [reply config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml:907]: `gcrane copy gcr.io/{k8s-testimages,k8s-staging-test-infra}/benchmarkjunit:latest`

**Pull request merged**